pallets/portfolio/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
+    IterableStorageDoubleMap,
+};
+use frame_system::{self as system, ensure_signed};
+use polymesh_common_utilities::{identity::Trait as IdentityTrait, CommonTrait, Context};
+use polymesh_primitives::{IdentityId, PortfolioId, PortfolioName, PortfolioNumber, Ticker};
+use sp_arithmetic::traits::Saturating;
+use sp_std::{convert::TryFrom, prelude::Vec};
+
+type Identity<T> = pallet_identity::Module<T>;
+
+/// The ticker and balance of an asset to be moved from one portfolio to another.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct MovePortfolioItem<Balance> {
+    /// The asset ticker.
+    pub ticker: Ticker,
+    /// The asset balance.
+    pub amount: Balance,
+}
+
+impl<Balance> Default for MovePortfolioItem<Balance>
+where
+    Balance: Default,
+{
+    fn default() -> MovePortfolioItem<Balance> {
+        MovePortfolioItem {
+            ticker: Ticker::default(),
+            amount: Balance::default(),
+        }
+    }
+}
+
+pub trait Trait: CommonTrait + IdentityTrait {
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Session {
+        /// The set of existing portfolios with their names. If a certain pair of a DID and
+        /// portfolio number maps to `None` then such a portfolio doesn't exist. Conversely, if a
+        /// pair maps to `Some(name)` then such a portfolio exists and is called `name`.
+        pub Portfolios get(fn portfolios):
+            double_map hasher(blake2_128_concat) IdentityId, hasher(twox_64_concat) PortfolioNumber =>
+            Option<PortfolioName>;
+        /// Asset balances of portfolios.
+        pub PortfolioAssetBalances get(fn portfolio_asset_balances):
+            double_map hasher(blake2_128_concat) PortfolioId, hasher(blake2_128_concat) Ticker =>
+            T::Balance;
+        /// The next portfolio sequence number of an identity.
+        pub NextPortfolioNumber get(fn next_portfolio_number):
+            map hasher(blake2_128_concat) IdentityId => PortfolioNumber;
+    }
+}
+
+decl_event! {
+    pub enum Event<T> where
+        Balance = <T as CommonTrait>::Balance,
+    {
+        /// The portfolio has been successfully created.
+        PortfolioCreated(IdentityId, PortfolioNumber, PortfolioName),
+        /// The portfolio has been successfully removed.
+        PortfolioDeleted(IdentityId, PortfolioNumber),
+        /// A token amount has been moved from one portfolio to another. `None` denotes the default
```

Comment:
I appreciate your attention to how doc comments look in the code. My approach is to format them to the conventional 100-character line width in the editor. Adding or removing words is better handled with the automated formatting than with the semantic one because automated tools don't recognise semantic line breaks.

If it's very important to you, we can start using semantic line breaks. My impression however that this is a very subtle convention that is not going to be universally followed and is going to require periodic maintenance.

Reply:
Ah; if you're using automated tooling to format these then they'll indeed be broken quickly.

This isn't at all important to me, but I figured I'd mention it in case you'd hadn't considered & might adopt it as a habit (as I eventually did upon a recommendation by someone else). My approach is to use semantic linefeeds in new code or if I'm substantially changing the text of a doc comment, but it wouldn't occur to me to periodically maintain doc comments like that. Basically, if you apply semantic linefeeds, then great, if not, then no big deal!